### ISSUE TITLE
Ad container shown on Le Bon Coin

### ISSUE THREAD

**Issue by Silejonu** [AdguardTeam/AdguardFilters#143567]
### Prerequisites

- [X] This site DOES NOT contains sexually explicit material, otherwise use [NSFW-specific form](./new?template=bug_report_NSFW.yml);
- [X] Filters were updated before reproducing an issue;
- [X] AdGuard product version is up-to-date;
- [X] Browser version is up-to-date;
- [X] If the site or app is broken, disabling AdGuard protection resolves an issue.

### What product do you use?

Other ad blocker

### AdGuard version

uBlock Origin 1.47.0

### What type of problem have you encountered?

Missed ads or ad leftovers

### Which browser(s) do you use?

Firefox

### Which device do you use?

Desktop

### Where is the problem encountered?

https://www.leboncoin.fr/annonces/offres

### What filters do you have enabled?

AdGuard Base filter, AdGuard Mobile Ads filter, AdGuard Chinese filter, AdGuard French filter, AdGuard Tracking Protection filter, AdGuard URL Tracking filter, AdGuard Social Media filter, AdGuard Annoyances filter, EasyList, EasyList Germany, EasyPrivacy, ABPVN List, List-KR, EasyList Cookie List, Legitimate URL Shortener

### Add your comment and screenshots

Ads are blocked in the search page, but the ad container is still shown.

![Screenshot from 2023-02-21 18-56-36](https://user-images.githubusercontent.com/48565914/220424170-fa785755-e070-4bc9-a96b-983740fe576f.png)

### Privacy

- [X] I agree to follow this condition

**Comment by Silejonu** [comment]
The fix does not seem to be working on searches when a category is selected.

Example: https://www.leboncoin.fr/informatique/offres

**Comment by BlazDT** [comment]
Modified rule, should work after update in about 3-4 hours.

**Comment by Silejonu** [comment]
Sorry to bother you once again: the fix does not work for categories with tiles organised side-by-side.

There are also blank spaces where sponsored links are supposed to be (I imagine that's unavoidable).

It affects the following categories:
https://www.leboncoin.fr/_maison_/offres
https://www.leboncoin.fr/_mode_/offres
https://www.leboncoin.fr/_loisirs_/offres
https://www.leboncoin.fr/_animaux_/offres
https://www.leboncoin.fr/autres/offres

Screenshot of a visible container + two blank spaces where sponsored links are supposed to be:
![Screenshot from 2023-03-01 17-50-14](https://user-images.githubusercontent.com/48565914/222207451-4eb5eab2-0a54-465f-a0af-27e46e644e4f.png)

**Comment by BlazDT** [comment]
Empty elements in sorting requires JS rule to remove all advertising elements in css style `grid-template-areas` for `#mosaic_with_owner`

**Comment by Uriopass** [comment]
Hello, it seems like this rule also removes the "Urgent" classified ads which are not ads container/sponsored ads but normal ads with a small badge on them.
Here you can see I filter on "urgent" ads and they all disappear:
![image](https://user-images.githubusercontent.com/5420739/225035470-cc7b7fc6-33e0-4432-ab24-f7a7c186e007.png)

**Comment by Silejonu** [comment]
> Hello, it seems like this rule also removes the "Urgent" classified ads which are not ads container/sponsored ads but normal ads with a small badge on them. Here you can see I filter on "urgent" ads and they all disappear: ![image](https://user-images.githubusercontent.com/5420739/225035470-cc7b7fc6-33e0-4432-ab24-f7a7c186e007.png)

Indeed. Example URL: https://www.leboncoin.fr/recherche?category=38&urgent=1

**Comment by BlazDT** [comment]
Rule which hides it comes from Liste FR. So for hiding rule we only have to block the text `Sponsorisé`? First text `À la une` is also some kind of ad (translates to Featured)?

Please try this rule:
leboncoin.fr#@?#a[data-test-id]:-abp-has([class]:-abp-contains(/À la une|Sponsorisé|Urgent/))

**Comment by Silejonu** [comment]
> Rule which hides it comes from Liste FR. So for hiding rule we only have to block the text `Sponsorisé`? First text `À la une` is also some kind of ad (translates to Featured)?
> 
> Please try this rule: `leboncoin.fr#@?#a[data-test-id]:-abp-has([class]:-abp-contains(/À la une|Sponsorisé|Urgent/))`

Adding the line to "My filters" seems to fix the issue, indeed.

"À la une" should not be blocked. It is not an ad, but a service some people pay to give more visibility to their listing (much like "Urgent"). Only "Sponsorisé" posts are actual ads, and should be blocked.

**Comment by BlazDT** [comment]
So in theory working as intended when people have to pay to have such elements higher up in visibility.

**Comment by Uriopass** [comment]
"À la une" could be blocked but "urgent" ads do not have more visibility (in terms of ranking), they simply have a red outline and there's a fitler for them

**Comment by Silejonu** [comment]
> "À la une" could be blocked but "urgent" ads do not have more visibility (in terms of ranking), they simply have a red outline and there's a fitler for them

Blocking "À la une" would mean entirely blocking user-made listings from search results. They're not ads. They're just highlights. Blocking the highlight would somewhate make sense (though it's an unnecessary hassle in my opinion), but blocking the listings entirely significantly affects the user experience.

You can do another small test: disable your adblock, and favourite a "À la une" listing, then go check your favourites with "À la une" blocked. The listing will not be visible anymore. That is not desirable behaviour: this means some listings could disappear from one's favourites just because the seller has paid for "À la une" in the meantime.

**Comment by krystian3w** [comment]
You can see if in the html code the favorites list is different to the categories and search results.

In addition, how each type is twice somewhere in the list of categories and searches can also be cut out instead of eliminating the annoying label and border.

**Comment by zloyden** [comment]
@BlazDT I guess it shouldn't be blocked, added the rule.
@Silejonu update should be available in 2-3 hours, after that you can disable your rule.
